Bug: Polecat startup race condition causes Interrupted prompt

# Bug Report: Polecat startup race condition causes Interrupted prompt

## Summary

Polecats spawned via `gt sling` consistently hit the Claude Code "Interrupted" prompt immediately after startup, preventing autonomous work execution. This is caused by witness nudges arriving during Claude Code's API initialization phase.

## Environment

- **Gas Town version**: Current (as of 2026-01-13)
- **Claude Code version**: 2.1.6
- **OS**: macOS Darwin 25.2.0
- **Rig**: web

## Reproduction Steps

1. Start witness for a rig: `tmux attach -t gt-web-witness`
2. Spawn a polecat with work: `gt sling hq-xyz web`
3. Attach to polecat session: `tmux attach -t gt-web-<polecat>`
4. Observe: Session shows "Interrupted · What should Claude do instead?" prompt
5. Check hook status: `gt hook show web/<polecat>` → empty

## Root Cause Analysis

### Timeline from Debug Logs

From Claude Code debug log (`~/.claude/debug/*.txt`) for a typical polecat spawn:

```
17:12:34.377 [DEBUG] Watching for changes in setting files
17:12:34.628 [DEBUG] Getting matching hook commands for SessionStart
17:12:34.629 [DEBUG] Found 0 hook matchers in settings
17:12:35.162 [DEBUG] [API:request] Creating client
17:12:46.338 [DEBUG] Stream started - received first chunk
17:12:48.396 [DEBUG] Streaming aborted by user: Request was aborted
17:12:48.396 [ERROR] Error in non-streaming fallback: Request was aborted
```

**Key observations:**
- **17:12:34**: Claude Code starts, begins initialization
- **17:12:46**: API stream starts (SessionStart processing)
- **17:12:48**: Stream aborted "by user" (actually by witness nudge)
- **Duration**: Only **2 seconds** between API stream start and abort

### The Race Condition

When `gt sling` spawns a polecat, this sequence occurs:

1. `gt sling` creates worktree and starts tmux session with Claude Code
2. `gt sling` sends "▶ Start prompt sent" nudge immediately
3. Witness detects new polecat session
4. Witness sends "Work is on your hook. Run `gt hook` now..." nudge
5. **Race**: These nudges arrive while Claude Code is still initializing
6. Nudge triggers `UserPromptSubmit` event → aborts ongoing API stream
7. Result: "Interrupted" prompt instead of autonomous execution

### Why This Happens

Claude Code's initialization sequence takes ~10-15 seconds:
- Load settings and hooks
- Initialize API client
- Run SessionStart hooks
- Prepare for first user interaction

External messages (nudges) during this window create new user prompts, which abort the initialization API stream.

## Impact

**Severity**: High - Blocks all autonomous polecat work

**Affected operations:**
- All `gt sling` operations
- Witness → polecat communication
- Mayor → witness nudges (witness itself can hit this)
- Any agent receiving messages during first ~15 seconds of startup

**Current workarounds:**
- Manual UI intervention (attach to tmux, respond to Interrupted prompt)
- Wait 15+ seconds after spawn before sending first message
- Not automatable or scalable

## Evidence

### Sessions Affected

Multiple polecat spawns showed identical behavior:
- `web/valkyrie` (spawned 10:21:35, interrupted 10:21:37)
- `web/keeper` (spawned 10:22:03, interrupted 10:22:05)
- `web/morsov` (spawned 10:22:32, interrupted 10:22:34)
- `web/witness` (also stuck at Interrupted prompt from mayor nudge)

### Hook Status Mismatch

```bash
$ bd show hq-7ei | grep "Assignee\|HOOKED"
Assignee: web/polecats/valkyrie · HOOKED

$ gt hook show web/valkyrie
web/valkyrie: (empty)
```

Bead metadata shows HOOKED, but actual hook mechanism is empty because session never got past Interrupted prompt.

## Proposed Solutions

### Option 1: Built-in Spawn Delay (Recommended)

Add configurable delay in `gt sling` before sending start prompt:

```go
// internal/cmd/sling.go
type SlingConfig struct {
    StartupDelaySeconds int `yaml:"startup_delay_seconds"` // default: 15
}

func slingWork() {
    // ... create worktree, start session ...

    if cfg.StartupDelaySeconds > 0 {
        time.Sleep(time.Duration(cfg.StartupDelaySeconds) * time.Second)
    }

    // ... send start prompt ...
}
```

Configuration in `rigs.json` per-rig:
```json
{
  "rigs": {
    "web": {
      "startup_delay_seconds": 15
    }
  }
}
```

### Option 2: Claude Code Readiness Signal

Enhance Claude Code to signal when ready for input:
- Create `.claude/.ready` file after initialization completes
- `gt sling` polls for this file before nudging
- More robust but requires Claude Code changes

### Option 3: --no-nudge Flag

Add flag to disable automatic nudging:
```bash
gt sling hq-xyz web --no-nudge
# Polecat spawns but witness doesn't auto-nudge
# Witness waits for manual trigger or extended timeout
```

### Option 4: Witness Intelligence

Enhance witness to:
- Track polecat spawn timestamps
- Wait 15 seconds before first nudge to new polecats
- Only send nudges after startup grace period

## Related Issues

- Claude Code #11544: Hooks not loading from settings.json
- May be related to overall Claude Code initialization timing

## Additional Context

### Settings Hook Issue

Separate but related: Polecats don't find `.claude/settings.json` hooks because:
1. Settings must be at git worktree root
2. Polecats spawn at `<rig>/polecats/<name>/web/`
3. Shared settings at `<rig>/polecats/.claude/` aren't detected
4. Symlinks aren't followed

This compounds the startup issue but is a separate problem (settings discovery path).

## Testing Verification

After implementing fix, verify:

1. **Spawn 3 polecats in quick succession**:
   ```bash
   gt sling hq-001 web
   gt sling hq-002 web
   gt sling hq-003 web
   ```

2. **Wait 30 seconds, then check**:
   ```bash
   gt polecat list web
   # All should show active (not "done")

   gt hook show web/<each-polecat>
   # All should show hooked work
   ```

3. **Check tmux sessions**:
   ```bash
   for p in polecat1 polecat2 polecat3; do
     tmux capture-pane -t gt-web-$p -p | grep -i interrupt
   done
   # Should return empty (no Interrupted prompts)
   ```

## Recommendation

Implement **Option 1** (built-in spawn delay) as it:
- Requires no Claude Code changes
- Is configurable per-rig
- Solves the immediate problem
- Can be refined later with Options 2 or 4

Default delay of 15 seconds provides safety margin for Claude Code initialization across different system loads.

---

**Filed by**: Mayor role (autonomous coordinator)
**Date**: 2026-01-13
**Priority**: High (blocks autonomous agent operations)

--- a/internal/cmd/polecat_spawn.go
+++ b/internal/cmd/polecat_spawn.go
@@ -5,6 +5,7 @@ import (
 	"fmt"
 	"path/filepath"
 	"strings"
+	"time"
 
 	"github.com/steveyegge/gastown/internal/config"
 	"github.com/steveyegge/gastown/internal/constants"
@@ -144,6 +145,18 @@ func SpawnPolecatForSling(rigName string, opts SlingSpawnOptions) (*SpawnedPolec
 		}
 		if err := polecatSessMgr.Start(polecatName, startOpts); err != nil {
 			return nil, fmt.Errorf("starting session: %w", err)
+		}
+
+		// Wait for runtime to be fully ready before returning.
+		// This prevents callers from sending nudges during Claude's initialization
+		// phase, which would interrupt the agent and cause "Interrupted" prompts.
+		// Fixes: https://github.com/steveyegge/gastown/issues/470
+		runtimeConfig := config.LoadRuntimeConfig(r.Path)
+		sessionName := polecatSessMgr.SessionName(polecatName)
+		if err := t.WaitForRuntimeReady(sessionName, runtimeConfig, 30*time.Second); err != nil {
+			// Non-fatal: session started, just not fully initialized yet.
+			// Caller should handle this gracefully.
+			fmt.Printf("Warning: runtime may not be fully ready: %v\n", err)
 		}
 	}
 

--- a/internal/polecat/session_manager.go
+++ b/internal/polecat/session_manager.go
@@ -173,10 +173,12 @@ func (m *SessionManager) Start(polecat string, opts SessionStartOptions) error {
 
 	runtimeConfig := config.LoadRuntimeConfig(m.rig.Path)
 
-	// Ensure runtime settings exist in polecats/ (not polecats/<name>/) so we don't
-	// write into the source repo. Runtime walks up the tree to find settings.
-	polecatsDir := filepath.Join(m.rig.Path, "polecats")
-	if err := runtime.EnsureSettingsForRole(polecatsDir, "polecat", runtimeConfig); err != nil {
+	// Ensure runtime settings exist in polecat's home directory (polecats/<name>/).
+	// This keeps settings out of the git worktree while allowing runtime to find them
+	// when walking up the tree from workDir (polecats/<name>/<rigname>/).
+	// Each polecat gets isolated settings rather than sharing a single settings file.
+	polecatHomeDir := m.polecatDir(polecat)
+	if err := runtime.EnsureSettingsForRole(polecatHomeDir, "polecat", runtimeConfig); err != nil {
 		return fmt.Errorf("ensuring runtime settings: %w", err)
 	}
 

